src/Business Foundation/App/NoSeries/readme_refactoring.md

```diff
@@ -0,0 +1,133 @@
+# Refactoring to use the new No. Series module
+
+This document is created to give a brief explanation of how refactor your code to use the new No. Series module.
+Old refers to the current ways of using the No. Series implementation while New refers to using the newly created No. Series module.
+
+## Uptake examples
+
+### TryGetNextNo
+
+Old:
+```
+DocNo := NoSeriesMgt.TryGetNextNo(GenJnlBatch."No. Series", EndDateReq);
+```
+New:
+```
+DocNo := NoSeries.PeekNextNo(GenJnlBatch."No. Series", EndDateReq);
+```
+
+### GetNextNo with delayed modify
+
+Old:
+```
+if DocNo := NoSeriesMgt.GetNextNo(GenJnlBatch."No. Series", EndDateReq, false) then
+    NoSeriesMgt.SaveNoSeries();
+```
+New:
+You have two options here, either you can Peek the next No. and update or use batch
+```
+if DocNo := NoSeries.PeekNextNo(GenJnlBatch."No. Series", EndDateReq) then
+    NoSeries.GetNextNo(GenJnlBatch."No. Series", EndDateReq);
+```
+or
+```
+if DocNo := NoSeriesBatch.GetNextNo(GenJnlBatch."No. Series", EndDateReq) then
+    NoSeriesBatch.SaveState();
+```
+
+### InitSeries
+
+Now InitSeries is a very complex implementation. In most cases where InitSeries is used, we verify that the given No. is not set, which this example will also verify. If it is, you will need to verify whether manual Nos are allowed.
+
+Old:
+```
+if "No." = '' then begin
+    GLSetup.Get();
+    GLSetup.TestField("Bank Account Nos.");
+    NoSeriesMgt.InitSeries(GLSetup."Bank Account Nos.", xRec."No. Series", 0D, "No.", "No. Series");
+end;
+```
+New:
+```
+if "No." = '' then begin
+    GLSetup.Get();
+    GLSetup.TestField("Bank Account Nos.");
+    "No. Series" := GLSetup."Bank Account Nos.";
+    NoSeries.AreRelated(GLSetup."Bank Account Nos.", xRec."No. Series") then
+        "No. Series" := xRec."No. Series"
+    "No." := NoSeries.GetNextNo("No. Series");
+end;
+```
+The new style is a bit more lines but better describes what is happening. Furthermore to keep this backwards compatible with old events, please add calls to the obsoleted functions NoSeriesManagement.RaiseObsoleteOnBeforeInitSeries and NoSeriesManagement.RaiseObsoleteOnAfterInitSeries. Example:
+```
+if "No." = '' then begin
+    GLSetup.Get();
+    GLSetup.TestField("Bank Account Nos.");
+    NoSeriesManagement.RaiseObsoleteOnBeforeInitSeries(GLSetup."Bank Account Nos.", xRec."No. Series", 0D, "No.", "No. Series", IsHandled);
+    if not IsHandled then begin
+        "No. Series" := GLSetup."Bank Account Nos.";
+        NoSeries.AreRelated(GLSetup."Bank Account Nos.", xRec."No. Series") then
+            "No. Series" := xRec."No. Series"
+        "No." := NoSeries.GetNextNo("No. Series");
+        NoSeriesManagement.RaiseObsoleteOnAfterInitSeries("No. Series", GLSetup."Bank Account Nos.", 0D, "No.");
+    end;
+end;
+```
+
+### Document posting with delayed modify
```

Comment:
Should we introduce this scenario, or is the section title enough for the audience?